adds actionable button documentation (#2962)
opensrp/fhircore: docs/engineering/android-app/configuring/config-types/widget.mdx

secondaryTextActions | The list of actions associated with the secondary text. If the list is empty, no actions will be displayed | | |
overflow | Determines how the text should be truncated if it is too long to fit | | [TextOverflow.ELLIPSIS] |

> ##How to use ACTIONABLE BUTTON

Reviewer: ```suggestion
## How to use BUTTON
```

> Defines the configuration properties for a Actionable Button view, which is a view that displays button with text strings, it's status ,visibility mode, and if it is enabled. This view can also have actions associated with it.

Reviewer: ```suggestion
Defines the configuration properties of a Button view, which is a view that can do some actions when clicked. Additionally, the button can be styled using available properties such as `status`, `contentColor`, and `buttonType`.
```

 ``` json
 [
   {
     "viewType": "BUTTON",
     "smallSized": true,
     "text": "@{taskDescriptionWithStartDate}",
     "status": "@{taskStatusColorCode}",
     "visible": "true",
     "enabled": "true",
     "actions": []
   }
 ]
```

### Config properties of ACTIONABLE BUTTON

> |Property | Description | Required | Default |

Reviewer: might be a good time where we add another column on what possible values that a property can use.
one such example is the ServiceStatus issue.
`DUE` | `OVERDUE` | `COMPLETED`

Reviewer: I'm referring to the column headings

Reviewer: default is more of "if the property is not defined, use x as the value"

Reviewer: Oh, let me clarify, I mean column heading in this kind of way:

The column heading:
Property | Description | Required | Default | Values

The column below the "Values" heading would be:
DUE | OVERDUE | COMPLETED

Let me know if that makes sense

Reviewer: yes

|--|--|:--:|:--:|

> |viewType|The view type should be `BUTTON` it helps in knowing the kind of view at hand|Yes| |

Reviewer: ```suggestion
| viewType | The view type should be `BUTTON` it helps in knowing the kind of view at hand | Yes | |
```
same for others

|text|The text represents the button name that is what the button action name is e.g Add Patient|Yes| |

> |status| status is the current mode of the button it can be a pure text like `Started` or even the color code of different modes|Yes| |

Reviewer: not sure if `Started` is correct, I think we are using [ServiceStatus](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ServiceStatus.kt) for the type of status that the codebase support

> |visible| This is the state of the button that is if it should be shown or not|No| |

Reviewer: ```suggestion
| visible | The state whether a button should be shown or hidden | No | |
```

> |enabled| enabled is used to set if the button is active |Yes| "true"|

Reviewer: I think we need to add more clarity on what does it mean for a button to be "active"

secondaryTextActions | The list of actions associated with the secondary text. If the list is empty, no actions will be displayed | | |
overflow | Determines how the text should be truncated if it is too long to fit | | [TextOverflow.ELLIPSIS] |

##How to use BUTTON
Defines the configuration properties of a Button view, which is a view that can do some actions when clicked. Additionally, the button can be styled using available properties such as `status`, `contentColor`, and `buttonType`
 ``` json
 [
   {
     "viewType": "BUTTON",
     "smallSized": true,
     "text": "@{taskDescriptionWithStartDate}",
     "status": "@{taskStatusColorCode}",
     "visible": "true",
     "enabled": "true",
     "actions": []
   }
 ]
```

### Config properties of BUTTON

> |Property | Description | Required | Default | Value |

Reviewer 2: Perhaps **Values** in plural might be more appropriate

Author: Fixing it right away @ndegwamartin

|--|--|:--:|:--:|:--:|
viewType | The view type should be `BUTTON` it helps in knowing the kind of view at hand | Yes | | |
text | The text represents the button name that is what the button action name is e.g Add Patient | Yes | | |
status | status indicate the service Status of the action that the button is supposed to perform this service status | Yes | | `DUE`, `OVERDUE`, `UPCOMING`,`COMPLETED`, `IN_PROGRESS`, `EXPIRED` |
visible | The state whether a button should be shown or hidden | No | |
enabled | enabled is used to set if the button is active that is if a button is set to perform the required action by the user | Yes | "true" | |
clickable | clickable is used to set if the button is allowed to perform its action on click or not | No | | |

> smallSized | smallSized determines if the button needs to be shown small size  | No | | |

Author: @FikriMilano I include them since they were among the`ButtonProperties` should I just remove them?

> smallSized | smallSized determines if the button needs to be shown small size  | No | | |

Reviewer: Can you give me a link to that code? I couldn't find it

> smallSized | smallSized determines if the button needs to be shown small size  | No | | |

Author: I had it on my local repo I think the configuration I was referring to is outdated

actions | This is a list of operation that the button does when it is clicked | Yes | | |
backgroundColor |The background color of the view, specified as a string in the format "#RRGGBB" or "#AARRGGBB". If this property is null, the view will use its parent's background color. | No |  Null | |
padding | Offsets the content of the view by a specific number of pixels. This should be a number | No | 0 | |
alignment | The alignment of the view within's its parent layout. This value is used to distribute the available space among multiple views | | [ViewAlignment.None]| |
fillMaxWidth | Determines whether the view should fill its parent's width | | true | |
fillMaxHeight | Determines whether the view should fill its parent's height | | false | |
fontSize | The font size of the primary and secondary button, in float |  | 14.0f | |
borderRadius | Rounds the corners of the view's outer edge | | 14 | |
startIcon | startIcon is used to specify an icon within the button if it is needed by default it is always null |  | null | |

> letterSpacing | letterSpacing is used to specify the amount of spacing that should be between the button text |  | 0 | |

Reviewer: ```suggestion
letterSpacing | The amount of space to be added between each letter of the button's text |  | 0 | |
```
